pkg/build/build.go

```diff
@@ -173,3 +181,146 @@ func CreateTarball(source, target string) error {
 			return err
 		})
 }
+
+// getImageTag get tag name from image name
+// if no tag is specified return 'latest'
+func GetImageTag(image string) string {
+	// format:      registry_host:registry_port/repo_name/image_name:image_tag
+	// example:
+	// 1)     myregistryhost:5000/fedora/httpd:version1.0
+	// 2)     myregistryhost:5000/fedora/httpd
+	// 3)     myregistryhost/fedora/httpd:version1.0
+	// 4)     myregistryhost/fedora/httpd
+	// 5)     fedora/httpd
+	// 6)     httpd
+	imageAndTag := image
+
+	imageTagSplit := strings.Split(image, "/")
+	if len(imageTagSplit) >= 2 {
+		imageAndTag = imageTagSplit[len(imageTagSplit)-1]
+	}
+
+	p := strings.Split(imageAndTag, ":")
+	if len(p) == 2 {
+		return p[1]
+	}
+	return "latest"
+}
+
+func GetImageName(image string) string {
+	imageAndTag := image
+
+	imageTagSplit := strings.Split(image, "/")
+	if len(imageTagSplit) >= 2 {
+		imageAndTag = imageTagSplit[len(imageTagSplit)-1]
+	}
+	p := strings.Split(imageAndTag, ":")
+	if len(p) <= 2 {
+		return p[0]
+	}
+
+	return image
+}
+
+func BuildS2I(image, context, builderImage string) error {
+
+	labels := map[string]string{
+		"build": GetImageName(image),
```

Comment:
it would be nice to have `build` as constant in `spec` package similar to other labels and annotations
(`appLabelKey`,`appVersion`)